### ISSUE TITLE
Need object for working with 2d tables as in exel

### ISSUE THREAD

**Issue by grabanton** [porres/pd-else#917]

**Comment by porres** [comment]
ok, please provide more details on what you're looking for and why jmmmp/cellblocks doesn't do it for you...

please also tell why exactly you need this for and what you want to achieve as the final goal

**Comment by grabanton** [comment]
Oh, I'm sorry! I just read it.
The main problem with jmmmp/cellblocks is that there is no way to edit the contents of the cells through the GUI.
I want to use this object to implement a custom markov chain. I would like to be able to conveniently edit the weight table.

**Comment by grabanton** [comment]
You can make an object that will return a list of values ​​on a bang, where the first two elements are the size of the table vertically and horizontally. Also, of course, you need to be able to use commands to set values ​​by address and pick them up.

**Comment by porres** [comment]
> I want to use this object to implement a custom markov chain.

where is ths markv chain patch? I wanna have a look, do you know about cyclone/prob?

**Comment by grabanton** [comment]
Yes, I know about probe. But I want to implement an algorithm from the csound patch where the Markov chain itself is an abstraction that simply returns the next array index. And an array can contain absolutely any data. This is very convenient for generating events with different frequencies, amplitudes and durations using just one Markov chain object. Moreover, you can change both arrays of values ​​and tables of weights on the fly. To implement the algorithm itself, I will use ophelia.
[markov_example.zip](https://github.com/porres/pd-else/files/5898134/markov_example.zip)
This patch doesn't sound very interesting, but it illustrates the idea well.

**Comment by grabanton** [comment]
Here my example patch builded in Pd )
[markov.zip](https://github.com/porres/pd-else/files/5904157/markov.zip)
